auditlog/models/rule.py

```diff
@@ -237,28 +239,31 @@ def _make_read(self):
         self.ensure_one()
         log_type = self.log_type
 
-        def read(self, *args, **kwargs):
-            result = read.origin(self, *args, **kwargs)
+        @api.multi
+        def read(self, fields=None, load='_classic_read'):
+            result = read.origin(self, fields, load)
             # Sometimes the result is not a list but a dictionary
             # Also, we can not modify the current result as it will break calls
             result2 = result
             if not isinstance(result2, list):
                 result2 = [result]
             read_values = dict((d['id'], d) for d in result2)
             # Old API
-            if args and isinstance(args[0], sql_db.Cursor):
-                cr, uid, ids = args[0], args[1], args[2]
+            if self and isinstance(self.env.cr, sql_db.Cursor):
+                cr, uid, ids = self.env.cr, self.env.uid, self.ids
```

Comment:
This code is to be compatible with old API which can always be present in a Odoo 10.0 project (modules not standard nor in OCA), that's why the signature of the method was `*args, **kwargs` to inspect the content of `args`.
If your call to `read(...)` is made with the new API, the rule creation is already satisfied below line 264 ("New API", https://github.com/OCA/server-tools/pull/1259/files#diff-e2cdd2f4b9da04f93936a15fa53fef79R269).

Reply:
I can let the signature as *args, but the signature *args, **kwargs, was what was generating the error, fixed by this patch. The actual signature for read in Odoo 10.0, is:

def read(self, fields=None, load='_classic_read'): #Two parameters

I can let the signature like:
def read(self, *args):

So the next sentence doesn't make sense in this case: cr, uid, ids = args[0], args[1], args[2]

I guess read in Odoo 10.0, isn't compatible with old api anymore. The module is working.

Reply:
Okay, I'm testing a piece of code with an `read` overload in old API, and it seems it's not compatible anymore with the new API in Odoo 10.0.
So OK to drop the support for old API. But regarding the fix, the one proposed here is wrong. The `_make_read` method should be written like this:
```python

        @api.multi
        def read(self, fields=None, load='_classic_read'):
            result = read.origin(self, fields, load)
            read_values = dict((d['id'], d) for d in result)
            # If the call came from auditlog itself, skip logging:
            # avoid logs on `read` produced by auditlog during internal
            # processing: read data of relevant records, 'ir.model',
            # 'ir.model.fields'... (no interest in logging such operations)
            if self.env.context.get('auditlog_disabled'):
                return result
            self = self.with_context(auditlog_disabled=True)
            rule_model = self.env['auditlog.rule']
            rule_model.sudo().create_logs(
                self.env.uid, self._name, self.ids,
                'read', read_values, None, {'log_type': log_type})
            return result

        return read
```